code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -125,104 +126,85 @@
 	if(user.transferItemToLoc(I, src))
 		to_chat(user, "<span class='notice'>You add [I] to [src].</span>")
 		holdingitems[I] = TRUE
-		updateUsrDialog()
 		return FALSE
 
 /obj/machinery/reagentgrinder/ui_interact(mob/user) // The microwave Menu //I am reasonably certain that this is not a microwave
 	. = ..()
-	var/is_chamber_empty = FALSE
-	var/is_beaker_ready = FALSE
-	var/processing_chamber = ""
-	var/beaker_contents = ""
-	var/dat = ""
-
-	if(!operating)
-		for (var/i in holdingitems)
-			var/obj/item/O = i
-			processing_chamber += "\A [O.name]<BR>"
-
-		if (!processing_chamber)
-			is_chamber_empty = TRUE
-			processing_chamber = "Nothing."
-		if (!beaker)
-			beaker_contents = "<B>No beaker attached.</B><br>"
-		else
-			is_beaker_ready = TRUE
-			beaker_contents = "<B>The beaker contains:</B><br>"
-			var/anything = FALSE
-			for(var/datum/reagent/R in beaker.reagents.reagent_list)
-				anything = TRUE
-				beaker_contents += "[R.volume] - [R.name]<br>"
-			if(!anything)
-				beaker_contents += "Nothing<br>"
-
-		dat = {"
-	<b>Processing chamber contains:</b><br>
-	[processing_chamber]<br>
-	[beaker_contents]<hr>
-	"}
-		if (is_beaker_ready)
-			if(!is_chamber_empty && !(stat & (NOPOWER|BROKEN)))
-				dat += "<A href='?src=[REF(src)];action=grind'>Grind the reagents</a><BR>"
-				dat += "<A href='?src=[REF(src)];action=juice'>Juice the reagents</a><BR><BR>"
-			else if (beaker.reagents.total_volume)
-				dat += "<A href='?src=[REF(src)];action=mix'>Mix the reagents</a><BR><BR>"
-		if(length(holdingitems))
-			dat += "<A href='?src=[REF(src)];action=eject'>Eject the reagents</a><BR>"
-		if(beaker)
-			dat += "<A href='?src=[REF(src)];action=detach'>Detach the beaker</a><BR>"
-	else
-		dat += "Please wait..."
-
-	var/datum/browser/popup = new(user, "reagentgrinder", "All-In-One Grinder")
-	popup.set_content(dat)
-	popup.set_title_image(user.browse_rsc_icon(icon, icon_state))
-	popup.open(TRUE)
-	return
 
-/obj/machinery/reagentgrinder/Topic(href, href_list)
-	if(..())
-		return
-	var/mob/user = usr
-	if(!user.canUseTopic(src))
```

Comment:
might be overlooking something, but are there no user stat checks anywhere in this?

Reply:
`ui_interact` already handles this

Reply:
no it doesn't.

Reply:
likewise if you're handcuffed with it still open

Reply:
Fixed